In [1]:
import sys
sys.path.append("../")
sys.path.append('C:\Users\James Pino\PycharmProjects\Magine')
from IPython.display import display, HTML
%matplotlib inline
import pandas as pd
import networkx as nx
import magine.ontology.enrichment_tools as et
import magine.networks.visualization.notebook_tools as nt
import magine.networks.visualization.notebooks.view as view
from magine.networks.network_subgraphs import NetworkSubgraphs

<IPython.core.display.Javascript object>

# Exploring network using enrichment analysis

### This example uses the enrichment output we just obtained to explore the network.

First, lets load the molecular network and the enrichment output from the previous runs. 

In [2]:
enrichment_array = pd.read_csv('Data/all_cisplatin_out.csv.gz', index_col=0)
network = nx.read_gpickle('Data/cisplatin_based_network.p')
subgraph_gen = NetworkSubgraphs(network=network)

For this example, we will only look at the proteomics data. 

In [3]:
proteomics = enrichment_array[enrichment_array['category'].str.contains('proteomics')].copy()

For this example, we just want to look at "biological processes" descriptions, so we will limit our analysis to only databases with this type of information

In [4]:
print(proteomics['db'].unique())

process_dbs = [
        'GO_Biological_Process_2017',
        'Humancyc_2016',
        'Reactome_2016',
        'KEGG_2016',
        'NCI-Nature_2016',
        'Panther_2016',
        'WikiPathways_2016',
]


['GO_Biological_Process_2017' 'GO_Molecular_Function_2017'
 'GO_Cellular_Component_2017' 'KEGG_2016' 'NCI-Nature_2016' 'Panther_2016'
 'WikiPathways_2016' 'BioCarta_2016' 'Humancyc_2016' 'Reactome_2016'
 'KEA_2015' 'ChEA_2016' 'DrugMatrix' 'Drug_Perturbations_from_GEO_2014']


In [5]:
time_1_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='01hr', db=process_dbs)
display(time_1_hour_prot)

term_name  combined_score  \
112198  negative regulation of transcription, DNA-temp...       68.261978   
114511              Cell Cycle_Homo sapiens_R-HSA-1640170       36.507320   
114512  Interleukin-2 signaling_Homo sapiens_R-HSA-451927       30.220851   
112200  resolution of meiotic recombination intermedia...       29.557281   
114513  Interleukin-3, 5 and GM-CSF signaling_Homo sap...       29.303950   
114514       Cell Cycle, Mitotic_Homo sapiens_R-HSA-69278       27.727089   
114515  Interleukin receptor SHC signaling_Homo sapien...       27.242088   
114516  MAPK family signaling cascades_Homo sapiens_R-...       26.440058   
114517      Signaling by FGFR2_Homo sapiens_R-HSA-5654738       25.922377   
114518  Insulin receptor signalling cascade_Homo sapie...       25.791980   
114519        Signalling by NGF_Homo sapiens_R-HSA-166520       25.783068   
112204                               mitotic cytokinesis        24.798241   
114520  Signaling by Interleukins_Homo sapiens_R-HSA-4...       24.453083   
114521        Signaling by FGFR_Homo sapiens_R-HSA-190236       24.242959   
114522      Signaling by ERBB4_Homo sapiens_R-HSA-1236394       24.213749   
114523  GRB2 events in EGFR signaling_Homo sapiens_R-H...       23.558041   
114524  SOS-mediated signalling_Homo sapiens_R-HSA-112412       23.443518   
114525  SHC1 events in ERBB4 signaling_Homo sapiens_R-...       23.349893   
114526  FRS-mediated FGFR4 signaling_Homo sapiens_R-HS...       23.297381   
114527  RAF/MAP kinase cascade_Homo sapiens_R-HSA-5673001       23.256870   
114528  FRS-mediated FGFR2 signaling_Homo sapiens_R-HS...       23.230305   
114529  SHC1 events in EGFR signaling_Homo sapiens_R-H...       23.200806   
114530  FRS-mediated FGFR3 signaling_Homo sapiens_R-HS...       23.135163   
114531  FRS-mediated FGFR1 signaling_Homo sapiens_R-HS...       23.000551   
114532  ARMS-mediated activation_Homo sapiens_R-HSA-17...       22.682926   
114533  MAP2K and MAPK activation_Homo sapiens_R-HSA-5...       22.681718   
114534  Frs2-mediated activation_Homo sapiens_R-HSA-17...       22.483992   
114535   MAPK1/MAPK3 signaling_Homo sapiens_R-HSA-5684996       22.466059   
114092  XPodNet - protein-protein interactions in the ...       22.460249   
114536  Signalling to p38 via RIT and RIN_Homo sapiens...       22.384695   
...                                                   ...             ...   
114564  FCERI mediated MAPK activation_Homo sapiens_R-...       16.133934   
114565        Signaling by PDGF_Homo sapiens_R-HSA-186797       16.126944   
114094         EGFR1 Signaling Pathway_Mus musculus_WP572       15.618928   
114566        Signaling by VEGF_Homo sapiens_R-HSA-194138       15.517250   
113890  Validated targets of C-MYC transcriptional rep...       15.369523   
113891  Regulation of Telomerase_Homo sapiens_4dfe97ca...       14.719095   
114095                      Cell Cycle_Homo sapiens_WP179       14.706942   
114567  Deactivation of the beta-catenin transactivati...       14.531321   
113892  E2F transcription factor network_Homo sapiens_...       14.331688   
114568  Fc epsilon receptor (FCERI) signaling_Homo sap...       13.725990   
114569  Cytokine Signaling in Immune system_Homo sapie...       13.297514   
114570            Immune System_Homo sapiens_R-HSA-168256       12.588445   
114571     Innate Immune System_Homo sapiens_R-HSA-168249       12.566682   
114572   Developmental Biology_Homo sapiens_R-HSA-1266738       11.464822   
113893  Notch signaling pathway_Homo sapiens_88f83518-...       11.237926   
114573  SUMO E3 ligases SUMOylate target proteins_Homo...        9.932085   
114574                DNA Repair_Homo sapiens_R-HSA-73894        9.569204   
114575  Transcription-Coupled Nucleotide Excision Repa...        9.180480   
114576             SUMOylation_Homo sapiens_R-HSA-2990846        9.159169   
114577  Formation of Incision Complex in GG-NER_Homo s...        9.073334   
114578  Signaling by Rho GTPases_Homo sapien

Since terms across databases might be redundant ("Interleukin-3, 5 and GM-CSF signaling_Homo sapiens_R-HSA-512988" and "Interleukin receptor SHC signaling_Homo sapiens_R-HSA-912526" have nearly a full overlap of genes), we want to eliminate duplicate terms and focus on the most enriched. For that we use the Jaccard Index. It is implmented in the find_similar_terms function in enrichment_tools (et).

In [6]:
filtered_1hr = et.find_similar_terms(time_1_hour_prot, threshold=.7)
display(filtered_1hr)

Number of rows went from 89 to 25


term_name  combined_score  \
112198  negative regulation of transcription, DNA-temp...       68.261978   
114511              Cell Cycle_Homo sapiens_R-HSA-1640170       36.507320   
114512  Interleukin-2 signaling_Homo sapiens_R-HSA-451927       30.220851   
112200  resolution of meiotic recombination intermedia...       29.557281   
114519        Signalling by NGF_Homo sapiens_R-HSA-166520       25.783068   
112204                               mitotic cytokinesis        24.798241   
114092  XPodNet - protein-protein interactions in the ...       22.460249   
112205                                chloride transport        22.310453   
114543            Axon guidance_Homo sapiens_R-HSA-422475       20.509597   
114093  Retinoblastoma (RB) in Cancer_Homo sapiens_WP2446       17.166900   
114094         EGFR1 Signaling Pathway_Mus musculus_WP572       15.618928   
113890  Validated targets of C-MYC transcriptional rep...       15.369523   
113891  Regulation of Telomerase_Homo sapiens_4dfe97ca...       14.719095   
114095                      Cell Cycle_Homo sapiens_WP179       14.706942   
114567  Deactivation of the beta-catenin transactivati...       14.531321   
113892  E2F transcription factor network_Homo sapiens_...       14.331688   
114570            Immune System_Homo sapiens_R-HSA-168256       12.588445   
113893  Notch signaling pathway_Homo sapiens_88f83518-...       11.237926   
114573  SUMO E3 ligases SUMOylate target proteins_Homo...        9.932085   
114574                DNA Repair_Homo sapiens_R-HSA-73894        9.569204   
114578  Signaling by Rho GTPases_Homo sapiens_R-HSA-19...        8.899833   
114582  Unfolded Protein Response (UPR)_Homo sapiens_R...        8.076945   
114583  Processing of Capped Intron-Containing Pre-mRN...        7.805714   
114596  Platelet Aggregation (Plug Formation)_Homo sap...        5.492674   
114600  Apoptotic cleavage of cellular proteins_Homo s...        5.230790   

        adj_p_value  rank                                              genes  \
112198     0.001601     1  ARID5B,BCLAF1,BRCA1,CALR,CBY1,CHD8,ENO1,HDAC1,...   
114511     0.000267     1  ACD,AKAP9,BRCA1,CDC16,CDC20,CDC7,CLASP2,DCTN1,...   
114512     0.001586     2  AKAP9,BRAF,CNKSR2,CUL3,HAVCR2,INPPL1,IRS2,MAPK...   
112200     0.016485     3                             ERCC4,SLX4,TOP2A,TOP2B   
114519     0.001586     9  AKAP13,AKAP9,ARHGEF16,BRAF,CNKSR2,CUL3,HDAC1,I...   
112204     0.030953     7                      APC,CEP55,PDCD6IP,PLK1,SPTBN1   
114092     0.011088     1  APC,ARPC1A,BRAF,CAPZA1,CDH7,CLNK,EIF4ENIF1,IGF...   
112205     0.029255     8                       ANO4,ANO5,ANO6,CLIC4,SLC26A8   
114543     0.001586    33  AKAP9,ARPC1A,BRAF,CLASP2,CLTCL1,CNKSR2,CRMP1,C...   
114093     0.017626     2         CDC7,HDAC1,MYC,PRIM1,RRM1,SMC1A,TOP2A,TYMS   
114094     0.041761     3  APPL1,HDAC1,INPPL1,KRT18,MAP3K3,MAPK3,MYC,PLEC...   
113890     0.009573     1             BRCA1,HDAC1,MYC,NDRG1,NFYC,PPP2R4,RBL1   
113891     0.009573     2                ACD,HDAC1,MAPK3,MYC,NCL,XRCC6,ZNFX1   
114095     0.029875     4         APC,CDC20,CDC7,HDAC1,PLK1,RBL1,SMC1A,YWHAZ   
114567     0.001586    57                     APC,CBY1,CHD8,HDAC1,TLE3,YWHAZ   
113892     0.010065     3               BRCA1,HDAC1,MYC,RBL1,RRM1,TYMS,XRCC1   
114570     0.008988    60  AKAP9,ARPC1A,BRAF,CALR,CAPZA1,CDC16,CDC20,CNKS...   
113893     0.016905     4                    ENO1,HDAC1,MARK2,MYC,NCOR1,SPEN   
114573     0.013351    63         BRCA1,NUP188,NUP88,PIAS1,SMC1A,TOP2A,TOP2B   
114574     0.032067    64  BRCA1,CUL4A,CUL4B,ERCC4,ERCC5,PIAS1,SLX4,USP1,...   
114578     0.036040    68  AKAP13,ARHGAP12,ARHGEF16,ARPC1A,CDC20,CLASP2,K...   
114582     0.032067    72                     CALR,DCTN1,LMNA,NFYC,SHC1,TLN1   
114583     0.048395    73  DDX5,FIP1L1,HNRNPA1,HNRNPM,NUP188,NUP88,PRPF3,...   
114596     0.032067    86                                GP1BA,SHC1,TLN1,VWF   
114600     0.032067    90                    

Now we can explore the top hits, which has been slimmed from 89 to 33 terms. Generally this is where the expert knowledge comes in. However, a quick search with each term and search terms of you molecule of interest tend to be useful. 

The first hit is 'negative regulation of transcription', which means that something caused genes not to be transcribed. Cisplatin causes DNA damage, thus negative regulaton of transcription makes sense. So does top hit 2, 'Cell Cycle_Homo sapiens_R-HSA-1640170'. 

A quick search for 'Interleukin-2 signaling' and 'Cisplatin' __[link](https://www.google.com/search?rlz=1C1CHBD_enUS721US721&ei=KzNeWuCxBsfq_AaSgYuYDQ&q=Interleukin-2+signaling+cisplatin&oq=Interleukin-2+signaling+cisplatin&gs_l=psy-ab.3..35i39k1.8097.9052.0.9196.10.10.0.0.0.0.145.897.7j3.10.0....0...1c.1.64.psy-ab..3.2.218....0.TInUjcZY740)__ returns a paper titled "Cisplatin at clinically relevant concentrations enhances interleukin-2 synthesis by human primary blood lymphocytes." __[link](https://www.ncbi.nlm.nih.gov/pubmed/10211553)__


We can link the two together to see how once might regulate the other by looking at the molecular interactions.

In [7]:
shorten_names = {
    'Cell Cycle_Homo sapiens_WP179':'Cell Cycle',
    'DNA Repair_Homo sapiens_R-HSA-73894' : 'DNA Repair',
    'resolution of meiotic recombination intermediates ': 'Meiotic Recombination'
                }
renamed_1hr = filtered_1hr.copy()
renamed_1hr['term_name'] = renamed_1hr['term_name'].replace(shorten_names)
term_net_1, mol_net_1 = nt.create_subnetwork(shorten_names.values(), renamed_1hr, network, '1hr')

('DNA Repair', 12)
('Meiotic Recombination', 4)
('Cell Cycle', 8)
Looking for direct edges


In [8]:
view.display_graph(term_net_1)

In [9]:
view.display_graph(mol_net_1, True)

## Exploring other top hits from 1 hour

### Side effects of cisplatin
Chemotherapy-induced peripheral neuropathy. __[link](https://www.frontiersin.org/articles/10.3389/fnins.2017.00481/full)__
It is not well understood why cisplatin causes CIPN. Surprisely here, we see that Axon Guidance has a combined score of 20.5. 33 species are effected by cisplain that are linked with axon guidance. We are not neural experts and did not know that axon guidance was related to CIPN. Using MAGINE we were able to find ties between the two.

In [10]:
axon_guidance = filtered_1hr[filtered_1hr['term_name'] == 'Axon guidance_Homo sapiens_R-HSA-422475']['genes'].tolist()[0].split(',')
print(axon_guidance)

['AKAP9', 'ARPC1A', 'BRAF', 'CLASP2', 'CLTCL1', 'CNKSR2', 'CRMP1', 'CUL3', 'IRS2', 'MAPK3', 'MARK3', 'MET', 'NRCAM', 'PTPRA', 'RASGRF2', 'SDCBP', 'SHC1', 'SPTBN1', 'SPTBN2', 'SRGAP2', 'TLN1', 'VWF']


In [11]:
g = nt.find_neighbors_of_list(network, axon_guidance, up_stream=True, down_stream=False, max_dist=1)
# nt.render_graph(g)

In [12]:
chloride_transport = filtered_1hr[filtered_1hr['term_name'] == 'chloride transport ']['genes'].tolist()[0].split(',')
nt.find_neighbors_of_list(network, chloride_transport, max_dist=1, render=True)

# 6 hour time point

In [13]:
time_6_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='06hr', db=process_dbs)
slimmed = et.find_similar_terms(time_6_hour_prot)
display(slimmed.head(25))

Number of rows went from 265 to 141


term_name  combined_score  \
124451                    mRNA splicing, via spliceosome        87.396296   
126815                 mRNA processing_Mus musculus_WP310       57.331009   
124452  positive regulation of transcription from RNA ...       56.402442   
127321           Gene Expression_Homo sapiens_R-HSA-74160       55.510780   
124453                                      RNA splicing        50.639384   
124454                          mRNA export from nucleus        46.486449   
124455                                   rRNA processing        45.422599   
124456                               protein sumoylation        42.216525   
126341                RNA transport_Homo sapiens_hsa03013       41.821069   
126342                  Spliceosome_Homo sapiens_hsa03040       38.345407   
126816  XPodNet - protein-protein interactions in the ...       36.008199   
124457          cellular response to DNA damage stimulus        35.101473   
127323  Activation of the AP-1 family of transcription...       33.437382   
126817         EGFR1 Signaling Pathway_Mus musculus_WP572       33.347966   
124458                                    RNA processing        32.992801   
126818                 mRNA Processing_Homo sapiens_WP411       32.884630   
127325             SUMOylation_Homo sapiens_R-HSA-2990846       31.744151   
124459           regulation of cellular response to heat        31.308147   
127328              Cell Cycle_Homo sapiens_R-HSA-1640170       29.160810   
126819         TSLP Signaling Pathway_Homo sapiens_WP2203       29.017371   
124461                                  response to cAMP        28.208944   
126820      TGF-beta Signaling Pathway_Homo sapiens_WP366       27.518886   
124462  nuclear-transcribed mRNA catabolic process, no...       26.712485   
127331  Growth hormone receptor signaling_Homo sapiens...       26.482133   
124463                        osteoblast differentiation        26.038610   

         adj_p_value  rank                                              genes  \
124451  1.684701e-07     1  CPSF2,CPSF3,DHX15,DHX8,DHX9,EIF4A3,HNRNPA2B1,H...   
126815  6.868228e-10     1  ACIN1,BCLAF1,BRWD1,CPSF2,CPSF3,DDX20,DDX21,DHX...   
124452  6.010178e-03     2  AATF,ARHGEF2,ATF2,CCNL1,CEBPZ,EGFR,FOXK2,FUBP3...   
127321  5.921719e-09     1  AEBP2,ANP32A,ATF2,ATR,BMS1,BOP1,CPSF2,CPSF3,DA...   
124453  1.510552e-05     3  DHX15,DHX8,EIF4A3,HNRNPC,HNRNPH3,MAGOHB,NCBP1,...   
124454  2.371966e-05     4  CPSF2,CPSF3,EIF4A3,HNRNPA2B1,MAGOHB,NCBP1,NUP1...   
124455  3.304134e-04     5  BMS1,BOP1,DKC1,NOP56,NOP58,PES1,RPL12,RPL3,RPL...   
124456  6.080538e-05     6  BCL11A,NUP153,NUP188,NUP214,NUP35,NUP98,SENP2,...   
126341  4.495322e-08     1  ACIN1,DDX20,EIF2S2,EIF3A,EIF3G,EIF4A3,EIF4B,EI...   
126342  4.797907e-08     2  ACIN1,DHX15,DHX8,EIF4A3,HNRNPC,HNRNPK,HNRNPU,H...   
126816  6.722057e-06     2  ABCC3,ARPC1A,ARPC1B,CAMK2D,CDC42EP4,CTNND1,CTT...   
124457  2.392831e-03     7  AATF,ATF2,ATR,DTL,LYN,MCM7,MYC,RAD50,RAD9A,RIF...   
127323  1.176482e-05     3                  ATF2,JUN,MAPK1,MAPK14,MAPK3,MAPK9   
126817  1.160157e-05     3  ABI1,ASAP1,CTNND1,EGFR,ERRFI1,GAB2,GIT1,HIP1,J...   
124458  3.304134e-04     8  DDX20,DHX8,DHX9,DKC1,HNRNPH3,HNRNPK,HNRNPU,RBM...   
126818  6.722057e-06     4  CPSF2,CPSF3,DDX20,DHX15,DHX8,DHX9,HNRNPA2B1,HN...   
127325  1.176482e-05     5  HNRNPC,HNRNPK,NOP58,NUP153,NUP188,NUP214,NUP35...   
124459  6.655661e-04     9  ATR,CAMK2D,HSPA1A,MAPK1,MAPK3,NUP153,NUP188,NU...   
127328  1.764157e-04     8  ATR,CDC23,CEP131,DIDO1,DKC1,GORASP2,KIF2A,MAPK...   
126819  2.093567e-05     5  EIF4EBP1,GAB2,LYN,MAPK1,MAPK14,MAPK3,MAPK9,MYC...   
124461  5.029015e-04    11                           DGKQ,JUN,JUNB,JUND,STAT1   
126820  2.270499e-05     6  ATF2,JUN,JUNB,JUND,MAPK1,MAPK14,MAPK9,MET,MYC,...   
124462  9.010199e-03    12  EIF4A3,MAGOHB,NCBP1,RBM8A,RNPS1,RPL12,RPL3,RPL...   
127331  1.508107e-04    11              IRS1,IRS2,LYN,MAPK1,MAPK3,SH2B1,STAT1   
124463  5.052837e-03

In [14]:
shorten_names = {
    'protein sumoylation ':'protein sumoylation',
    'Activation of the AP-1 family of transcription factors_Homo sapiens_R-HSA-450341' : 'AP1 activation',
    'response to cAMP ' : 'response to cAMP',
#     'Axon guidance_Homo sapiens_R-HSA-422475': 'Axon Guidance',
#   'Interleukin-2 signaling_Homo sapiens_R-HSA-451927': 'IL2',
#      'resolution of meiotic recombination intermediates ': 'Meiotic Recombination'
                }
time_6_hour_prot['term_name'] = time_6_hour_prot['term_name'].replace(shorten_names)
term_net_6, mol_net_6 = nt.create_subnetwork(shorten_names.values(), time_6_hour_prot, network, '06hr', cytoscape_js=False)


('response to cAMP', 5)
('protein sumoylation', 12)
('AP1 activation', 6)
Looking for direct edges


In [15]:
view.display_graph(term_net_6)

In [16]:
view.render_graph(mol_net_6)

In [17]:
# nt.find_neighbors(g=mol_net_1, start='ATR', up_stream=False, down_stream=True, max_dist=1, render=True)
# nt.find_neighbors(g=mol_net_1, start='ATR', up_stream=True, down_stream=False, max_dist=3, render=True)

In [18]:
time_24_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='24hr', db=process_dbs)
slimmed = et.find_similar_terms(time_24_hour_prot, threshold=.5)
display(slimmed.head(25))

Number of rows went from 378 to 141


term_name  combined_score  \
137544                    mRNA splicing, via spliceosome       209.979563   
141153           Gene Expression_Homo sapiens_R-HSA-74160      102.139686   
137545                                   rRNA processing        96.343182   
137546           regulation of cellular response to heat        84.291259   
140582                 mRNA processing_Mus musculus_WP310       75.595732   
137547                          mRNA export from nucleus        73.912638   
141156              Cell Cycle_Homo sapiens_R-HSA-1640170       62.048920   
137549          positive regulation of apoptotic process        53.817781   
137550                                      RNA splicing        50.724093   
137552  regulation of alternative mRNA splicing, via s...       45.264258   
141166      Infectious disease_Homo sapiens_R-HSA-5663205       43.423305   
141170             SUMOylation_Homo sapiens_R-HSA-2990846       41.460201   
140059                RNA transport_Homo sapiens_hsa03013       38.738445   
137554                                   mRNA processing        38.279921   
141176                Apoptosis_Homo sapiens_R-HSA-109581       36.840898   
140584       Leptin signaling pathway_Homo sapiens_WP2034       34.391387   
140585  Integrated Breast Cancer Pathway_Homo sapiens_...       32.691245   
140586      EGF/EGFR Signaling Pathway_Homo sapiens_WP437       32.542271   
140587         EGFR1 Signaling Pathway_Mus musculus_WP572       31.687542   
140309  Regulation of Telomerase_Homo sapiens_4dfe97ca...       30.694159   
137562                          neutrophil degranulation        29.244334   
137563          cellular response to DNA damage stimulus        28.060164   
137565  positive regulation of protein export from nuc...       27.656797   
137566          negative regulation of apoptotic process        27.292580   
140588  Integrated Pancreatic Cancer Pathway_Homo sapi...       27.113830   

         adj_p_value  rank                                              genes  \
137544  3.357108e-21     1  CPSF1,CSTF1,CWC15,DDX42,EFTUD2,EIF4A3,FIP1L1,F...   
141153  1.220822e-18     2  AIMP2,ANP32A,ATF2,ATR,BNIP3L,BOP1,C2ORF49,CD3E...   
137545  4.915441e-10     2  BOP1,DDX49,DKC1,EBNA1BP2,FBL,HEATR1,IMP3,KRR1,...   
137546  2.769570e-11     3  ATR,BAG3,CAMK2D,CCAR2,GSK3B,HSF1,HSP90AA1,HSPB...   
140582  1.401438e-13     1  ACIN1,AKAP1,BCLAF1,CPSF1,CSTF1,DDX21,DDX4,DND1...   
137547  1.055205e-09     4  CPSF1,EIF4A3,FIP1L1,HNRNPA2B1,NCBP1,NUP107,NUP...   
141156  6.097046e-10     5  AKAP9,ATR,BANF1,CEP131,CNTRL,DIDO1,DKC1,DYNC1H...   
137549  5.123684e-05     6  AKAP13,ANO6,ARHGEF1,ARHGEF11,ARHGEF7,BAD,BCLAF...   
137550  2.169909e-06     7  CCAR2,CDK12,EFTUD2,EIF4A3,HNRNPC,HNRNPH3,KHSRP...   
137552  4.438842e-07     9  HNRNPA1,HNRNPL,RBM15B,RBM4,RBM8A,RBMX,SAP18,SF...   
141166  1.147292e-07    15  BANF1,CALR,CANX,CLTA,EEF2,HMGA1,HSP90AA1,ISG15...   
141170  1.127330e-08    19  HDAC7,HNRNPC,HNRNPK,NOP58,NSMCE2,NUP107,NUP153...   
140059  9.324305e-08     2  ACIN1,EIF4A3,EIF4B,EIF4EBP1,EIF4EBP2,EIF4G3,EI...   
137554  1.025094e-05    11  EFTUD2,HNRNPA1,HNRNPA2B1,HNRNPR,METTL3,SF3A1,S...   
141176  1.683367e-07    25  ACIN1,ADD1,BAD,CASP3,DBNL,DIABLO,DSP,GSN,LMNA,...   
140584  7.235247e-06     3  ACACA,BAD,EIF4EBP1,GSK3A,GSK3B,IRS1,KHDRBS1,MA...   
140585  9.264562e-06     4  ATR,BAD,BMPR2,CASP3,EGFR,FOSL1,GSK3A,IRS1,JUN,...   
140586  7.235247e-06     5  ARHGEF1,ASAP1,CAV2,DNM1,EGFR,EIF4EBP1,EPN1,EPS...   
140587  1.166188e-05     6  ASAP1,CAV2,CTNND1,DNM1,EGFR,EPN1,EPS15,GIT1,HT...   
140309  2.381163e-06     1  DKC1,EGFR,HNRNPC,HSP90AA1,JUN,MAPK1,MAPK3,MTOR...   
137562  2.072846e-02    19  ACLY,AHSG,ANO6,CAP1,CAPN1,CD59,DBNL,DSP,DYNC1H...   
137563  3.805248e-03    20  ATF2,ATR,BCL3,CDKN2AIP,GNL1,PARP1,RAD18,RAD50,...   
137565  9.937782e-04    22                      BAG3,EMD,GSK3B,TP53,TPR,YWHAE   
137566  2.068391e-02    23  ANXA4,BAG3,BCL3,BNIP3,BNIP3L,CASP3,EGFR,ERCC5,...   
140588  6.148038e-05

In [19]:
shorten_names = {
                 'cellular response to DNA damage stimulus ': 'DDR',
                 'negative regulation of apoptotic process ': 'negative regulation of apoptosis',
                 'Apoptosis_Homo sapiens_R-HSA-109581' : 'Apoptosis',
                }
renamed = time_24_hour_prot.copy()
time_24_hour_prot['term_name'] = time_24_hour_prot['term_name'].replace(shorten_names)
term_net_24, mol_net_24 = nt.create_subnetwork(shorten_names.values(), time_24_hour_prot, network, '24hr', cytoscape_js=False)

('DDR', 20)
('negative regulation of apoptosis', 27)
('Apoptosis', 26)
Looking for direct edges


In [20]:
view.display_graph(term_net_24)

In [21]:
view.display_graph(mol_net_24, add_parent=True )

In [22]:
time_48_hour_prot = et.filter_dataframe(proteomics, p_value=0.05, combined_score=0.0, category='proteomics_up', sample_id='48hr', db=process_dbs)
slimmed = et.find_similar_terms(time_48_hour_prot, threshold=.5)
display(slimmed.head(25))

Number of rows went from 420 to 167


term_name  combined_score  \
151592                    mRNA splicing, via spliceosome       234.079702   
151593                          neutrophil degranulation       180.858577   
155332           Gene Expression_Homo sapiens_R-HSA-74160       92.138380   
151594  mitochondrial ATP synthesis coupled proton tra...       90.680332   
155334   Metabolism of proteins_Homo sapiens_R-HSA-392499       82.654616   
154762                 mRNA processing_Mus musculus_WP310       81.175763   
151596                             membrane organization        75.662017   
151598                          mRNA export from nucleus        67.106534   
151599          negative regulation of apoptotic process        61.614403   
151600                             RNA metabolic process        57.796749   
151601                            platelet degranulation        57.718950   
155335                 Disease_Homo sapiens_R-HSA-1643685       56.883870   
154764        Electron Transport Chain_Homo sapiens_WP111       55.883408   
151602            ER to Golgi vesicle-mediated transport        52.615159   
155337               Hemostasis_Homo sapiens_R-HSA-109582       52.388225   
154765  XPodNet - protein-protein interactions in the ...       52.009261   
155338  Vesicle-mediated transport_Homo sapiens_R-HSA-...       50.489896   
154233  Protein processing in endoplasmic reticulum_Ho...       47.316584   
154234               Focal adhesion_Homo sapiens_hsa04510       46.294756   
151604                                      RNA splicing        44.601721   
154478  ErbB1 downstream signaling_Homo sapiens_30d605...       39.575858   
151605           regulation of cellular response to heat        35.933418   
151606                                   rRNA processing        35.377604   
154768  Integrated Pancreatic Cancer Pathway_Homo sapi...       35.351683   
151607                cellular protein metabolic process        35.341205   

         adj_p_value  rank                                              genes  \
151592  6.840193e-24     1  DDX39B,DDX42,DHX9,FUS,HNRNPA0,HNRNPA1,HNRNPA2B...   
151593  5.282489e-13     2  ACAA1,AHSG,ALDOA,ANO6,ANXA3,APEH,ATP6V0A1,CD44...   
155332  4.733666e-17     3  ANP32A,APEH,BAZ1B,BDP1,BOP1,CD3EAP,CD44,CDK12,...   
151594  2.328503e-12     3  ATP5A1,ATP5B,ATP5C1,ATP5D,ATP5F1,ATP5H,ATP5I,A...   
155334  2.497306e-15     5  APEH,APOA1,ARF4,ATP5A1,ATP5B,CALR,CANX,CD59,CK...   
154762  3.223967e-15     1  ACIN1,BCLAF1,DDX21,DDX39B,DHX9,FBL,HNRNPA0,HNR...   
151596  7.658452e-07     5  AAK1,ACTR3,AP2B1,ARPC1A,CCDC88A,CTTN,DENND4C,D...   
151598  4.281073e-09     7  DDX39B,HNRNPA2B1,MAGOHB,NCBP1,NDC1,NUP153,NUP1...   
151599  7.827425e-05     8  ALB,ANXA4,ANXA5,ARF4,ARL6IP1,BAG3,CD44,DNAJA1,...   
151600  2.042800e-09     9  DDX54,HNRNPA0,HNRNPA1,HNRNPA2B1,HNRNPA3,HNRNPC...   
151601  2.573875e-07    10  ACTN1,ACTN4,AHSG,ALB,ALDOA,ANXA5,APOA1,APOH,CD...   
155335  2.740126e-09     6  ALB,AP2B1,APC,BAD,BANF1,CALR,CANX,CD9,CHMP7,CT...   
154764  6.727001e-11     3  ATP5A1,ATP5B,ATP5C1,ATP5D,ATP5F1,ATP5H,ATP5I,A...   
151602  3.217147e-06    11  ARF4,ATL2,BCAP31,CD59,COPA,COPG2,DYNC1LI1,ERGI...   
155337  1.783685e-09     8  ACTN1,ACTN4,AHSG,ALB,ALDOA,APOA1,APOH,ATP1B1,B...   
154765  2.495023e-09     4  ACTN4,ACTR3,APC,ARPC1A,ATP6V0A1,BAD,CD47,CDK4,...   
155338  3.569055e-09     9  ALB,APOA1,APOE,ARF4,CALR,CD59,CHMP7,COPA,COPG2...   
154233  4.097873e-10     2  BCAP31,CALR,CANX,CAPN2,CKAP4,DAD1,DDOST,DERL1,...   
154234  1.486071e-09     3  ACTN1,ACTN4,BAD,CAPN2,CAV2,CRKL,CTNNB1,DIAPH1,...   
151604  8.424171e-06    13  CDK12,DDX39B,HNRNPC,HNRNPH3,KHSRP,MAGOHB,NCBP1...   
154478  5.383828e-08     1  ACTR3,ARF4,BAD,CAPN2,EGFR,EPS8,JUN,MAPK1,MAPK3...   
151605  6.080024e-05    14  BAG3,HSBP1,HSPA8,HSPH1,MAPK1,MAPK3,NDC1,NUP153...   
151606  5.070748e-04    15  BOP1,DKC1,FBL,NOL11,NOP56,NOP58,RIOK3,RPL10,RP...   
154768  1.108409e-06     7  APC,BAD,BRCA2,CDK4,CTNNB1,DCC,DFFA,EGFR,FAS,FG...   
151607  7.286813e-04

In [23]:
shorten_names = {
                 'membrane organization ': 'Membrane Organization',
                 'negative regulation of apoptotic process ': 'negative regulation of apoptosis',
                 'platelet degranulation ' : 'platelet degranulation',
                }
time_48_hour_prot['term_name'] = time_48_hour_prot['term_name'].replace(shorten_names)
term_net_48, mol_net_48 = nt.create_subnetwork(shorten_names.values(), time_48_hour_prot, network, '48hr', cytoscape_js=False)

('negative regulation of apoptosis', 34)
('platelet degranulation', 23)
('Membrane Organization', 31)
Looking for direct edges


In [24]:
view.display_graph(term_net_48)

In [25]:
view.display_graph(mol_net_48, True)